gawk: 4.2.1 -> 5.0.1

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/gawk/versions.



meta.description for gawk is: '"GNU implementation of the Awk programming language"'.


<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- /bin/sh passed the binary check.
- 1 of 1 passed binary check by having a zero exit code.
- 0 of 1 passed binary check by having the new version present in output.
- found 5.0.0 with grep in /nix/store/bhrdx0n2afrmr0xf194am9qbg57sw1jh-gawk-5.0.0
- directory tree listing: https://gist.github.com/ee9ae1533fa01bbc0755ec61b956c68e
- du listing: https://gist.github.com/d83d21aeab87bfd8c84cca39b72169d9

</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

94296 total rebuild path(s)

27138 package rebuild(s)

26983 x86_64-linux rebuild(s)
24809 i686-linux rebuild(s)
18582 x86_64-darwin rebuild(s)
23922 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
AMB-plugins
AgdaSheaves
AgdaStdlib
CoinMP
DisnixWebService
EBTKS
EmptyEpsilon
FIL-plugins
Fabric
LASzip
Literate
MMA
NSPlist
OVMF
OVMF-CSM
PlistCpp
QuadProgpp
R
SDL
SDL2
SDL2_gfx
SDL2_image
SDL2_mixer
SDL2_net
SDL2_ttf
SDL_gfx
SDL_image
SDL_mixer
SDL_net
SDL_sixel
SDL_sound
SDL_stretch
SDL_ttf
SkypeExport
TotalParserCombinators
Xaw3d
_0x0
_1password
_2048-in-terminal
_20kly
_2bwm
_389-ds-base
_90secondportraits
_915resolution
_9pfs
a2jmidid
a2ps
a52dec
aacgain
aacskeys

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/bhrdx0n2afrmr0xf194am9qbg57sw1jh-gawk-5.0.0 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A gawk https://github.com/r-ryantm/nixpkgs/archive/569249f0848acccffe92d396a948e7978db0b9c3.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/bhrdx0n2afrmr0xf194am9qbg57sw1jh-gawk-5.0.0
ls -la /nix/store/bhrdx0n2afrmr0xf194am9qbg57sw1jh-gawk-5.0.0/bin
```


</details>
<br/>

r-ryantm committed:
gawk: 4.2.1 -> 5.0.1

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/gawk/versions

hedning commented:
NEWS: https://git.savannah.gnu.org/cgit/gawk.git/tree/NEWS

Changes from 4.2.1 to 5.0.0
---------------------------

1. Support for the POSIX standard %a and %A printf formats has been added.

2. The test infrastructure has been greatly improved, simplifying the
   contents of test/Makefile.am and making it possible to generate
   pc/Makefile.tst from test/Makefile.in.

3. The regex routines have been replaced with those from GNULIB, allowing
   me to stop carrying forward decades of changes against the original
   ones from GLIBC.

4. Infrastructure upgrades: Bison 3.3, Automake 1.16.1, Gettext 0.19.8.1,
   makeinfo 6.5.

5. The undocumented configure option and code that enabled the use of
   non-English "letters" in identifiers is now gone.

6. The `--with-whiny-user-strftime' configuration option is now gone.

7. The code now makes some stronger assumptions about a C99 environment.

8. PROCINFO["platform"] yields a string indicating the platform for
   which gawk was compiled.

9. Writing to elements of SYMTAB that are not variable names now
   causes a fatal error. THIS CHANGES BEHAVIOR.

10. Comment handling in the pretty-printer has been reworked almost completely
    from scratch. As a result, comments in many corner cases that were
    previously lost are now included in the formatted output.

11. Namespaces have been implemented! See the manual.  One consequence of this
    is that files included with -i, read with -f, and command line program
    segments must all be self-contained syntactic units. E.g., you can no
    longer do something like this:

   	gawk -e 'BEGIN {' -e 'print "hello" }'

12. Gawk now uses the locale settings for ignoring case in single byte
    locales, instead of hardwiring in Latin-1.

13. A number of bugs, some of them quite significant, have been fixed.
    See the ChangeLog for details.

tomberek reviewed (approved):
Ran this through my AdventOfCode solutions in awk, no problems found, but this is not a conclusive test due to the sheer number of reverese dependencies and being a system-level dependency. There are some breaking changes, so recommend some NixOS tests.

nixos-discourse commented:
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/12

disassembler commented:
@GrahamcOfBorg 	test installer.simple

lheckemann commented:
@GrahamcOfBorg build libgpg-error

This seems to fail

mmahut commented:
@GrahamcOfBorg build libgpgerror

mmahut commented:
ping @Fuuzetsu @vrthra about the libgpgerror failure.

lheckemann commented:
@Fuuzetsu @vrthra (libgpgerror maintainers) any chance you'd be able to look at this?

Commit:
libgpg-error: fix build with gawk 5

Cherry picked commit 6fe2cdd03c98270b0ce0026acf8dab0a6c4d3d63 from
https://github.com/NixOS/nixpkgs/pull/67801.

peti commented:
@GrahamcOfBorg build libgpgerror

disassembler commented:
cleaned up commit history, merging into staging.
